static/app/components/forms/fieldFromConfig.tsx

```diff
@@ -32,74 +34,67 @@ type Props = {
   noOptionsMessage?: () => string;
   onBlur?: (value, event) => void;
   stacked?: boolean;
-};
+}
 
-export default class FieldFromConfig extends Component<Props> {
-  render() {
-    const {field, ...otherProps} = this.props;
+function FieldFromConfig(props: FieldFromConfigProps): React.ReactElement | null {
+  const {field, ...otherProps} = props;
 
-    const props = {
-      ...otherProps,
-      ...field,
-    };
+  const componentProps = {
+    ...otherProps,
+    ...field,
+  };
 
-    switch (field.type) {
-      case 'separator':
-        return <FieldSeparator />;
-      case 'secret':
-        return <InputField {...props} type="password" />;
-      case 'range':
-        // TODO(ts) The switch on field.type is not resolving
-        // the Field union for this component. The union might be 'too big'.
-        return <RangeField {...(props as any)} />;
-      case 'blank':
-        return <BlankField {...props} />;
-      case 'bool':
-      case 'boolean':
-        return <BooleanField {...props} />;
-      case 'email':
-        return <EmailField {...props} />;
-      case 'hidden':
-        return <HiddenField {...props} />;
-      case 'string':
-      case 'text':
-      case 'url':
-        if (props.multiline) {
-          return <TextareaField {...props} />;
-        }
-        return <TextField {...props} />;
-      case 'number':
-        return <NumberField {...props} />;
-      case 'textarea':
-        return <TextareaField {...props} />;
-      case 'choice':
-      case 'select':
-      case 'array':
-        return <SelectField {...props} />;
-      case 'choice_mapper':
-        // TODO(ts) The switch on field.type is not resolving
-        // the Field union for this component. The union might be 'too big'.
-        return <ChoiceMapperField {...(props as any)} />;
-      case 'radio':
-        const choices = props.choices;
-        if (!Array.isArray(choices)) {
-          throw new Error('Invalid `choices` type. Use an array of options');
-        }
-        return <RadioField {...props} choices={choices} />;
-      case 'table':
-        // TODO(ts) The switch on field.type is not resolving
-        // the Field union for this component. The union might be 'too big'.
-        return <TableField {...(props as any)} />;
-      case 'project_mapper':
-        return <ProjectMapperField {...props} />;
-      case 'sentry_project_selector':
-        return <SentryProjectSelectorField {...props} />;
-      case 'select_async':
-        return <SelectAsyncField {...props} />;
-      case 'custom':
-        return field.Component(props);
-      default:
-        return null;
-    }
+  switch (field.type) {
+    case 'separator':
+      return <FieldSeparator />;
+    case 'secret':
+      return <InputField {...(componentProps as InputFieldProps)} type="password" />;
+    case 'range':
+      return <RangeField {...(componentProps as RangeFieldProps)} />;
```

Comment:
now we have a type 👏

Reply:
Yeah, but type casting here basically makes it useless :/